eng/build.yml

```diff
@@ -185,10 +175,9 @@ jobs:
         -configuration ${{ parameters.configuration }}
         -architecture ${{ parameters.architecture }}
         $(_CrossBuildArgs)
-        $(_TestArgs)
```

Comment:
The test parallelism issue due to using msbuild to run the tests should be controllable by using `ProjectToBuild` items with the `BuildInParallel` metadata set to false if we break out build from test (we probably want to still allow parallel builds). See https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L42 and https://github.com/dotnet/arcade/blob/main/src/Microsoft.DotNet.Arcade.Sdk/tools/Build.proj#L69. These would need to be only set when we know we are testing on the build machines or maybe controlled via a property that the yaml sets by default which would allow us to attain the same behavior locally if desired.